Automattic/jetpack
# Reconfigure eslint to pull from `.gitignore` and per-dir `.eslintignore`

#### Changes proposed in this Pull Request:

Unfortunately we have to do it in a somewhat hacky manner: in eslintrc
we run a function that reads all the ignore files and dumps the results
into eslint's `ignorePatterns` config.

Then we have to do it all over again for any sub-eslintrc that does
`root: true` or that extends any eslintrc that has the hack. Inheritance
is ok, but apparently we don't do that all that often (sigh).

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* The files linted by this versus current master on a clean checkout should be the same, other than the added/renamed files.
* The lints reported (for excluded files) should also be the same, other than the two in eslintrc files that are fixed in this PR.
* Do we want to hack the config like this for the convenience of no longer having to copy everything from `.gitignore` to a root-level `.eslintignore`?
